### ISSUE TITLE
record section options for shifting seismograms in time

### ISSUE THREAD

**Issue by carltape** [adjtomo/pysep#92]
Currently the record section tool allows for a move out velocity. For global studies, such as aligning on P wave arrival times, it would be optimal to have a user-specified time shift for each trace (since a single move-out velocity will not work well). In many cases, this might be the tau-P arrival time for P, which we already have.

Here's an excerpt from a separate code where each seismogram has been shifted by a time that is shown in the text label:
<img width="736" alt="Screen Shot 2023-03-01 at 4 19 07 PM" src="https://user-images.githubusercontent.com/4455456/222306013-29dbf274-652e-4245-8e80-79ac8b783c73.png">

This was the tshift variable in plotw_rs.m:
https://github.com/uafgeotools/GEOTOOLS/blob/master/matlab_util/util_data_syn/plotw_rs.m#L13
They key point was that tshift could be a vector of times, the same length as the number of seismograms.

This capability is used in GEOS 626 for examining the P wave train from 2004 Mw 9.2 Sumatra-Andaman.

**Comment by carltape** [comment]
This shows an approximation for how the P wave apparent velocity changes with epicentral distance:
<img width="679" alt="Screen Shot 2023-03-01 at 4 32 27 PM" src="https://user-images.githubusercontent.com/4455456/222307800-7192787d-8558-4cc0-bf0f-4c61d6a0d755.png">
So a single move out velocity (say 10.5 km/s) would be a rough approximation over these distances, but allowing for a variable to specify the time shift (or apparent velocity) would be better.

**Comment by bch0w** [comment]
Just looked at RecordSection and it can take a list of values for the parameter `time_shift_s`, so currently the capability is there to manually input time shifts related to e.g., P-arrival times

https://github.com/adjtomo/pysep/blob/3f5d7ab12aee663abe907036ca6eb757879d3722/pysep/recsec.py#L234-L241

I agree, would be useful to also allow `time_shift_s` to be something like 'p_phase' and have RecordSection search the TauP header for  the corresponding arrival time.

**Comment by bch0w** [comment]
A quick workaround in-code would be to grab the P-arrivals manually and feed that in as a list, something like:

```python
time_shift_s = [tr.stats.sac['t5'] for tr in st]  # 't5' is P-wave arrival time
recsec = RecordSection(st=st, time_shift_s=time_shift_s ...)
```

**Comment by carltape** [comment]
Sorry for my misunderstanding. Great that the capabilities are there. We'll try alignment on the P times from tau-P.

**Comment by bch0w** [comment]
This has been addressed in #94, see example below of shifting on the P arrival. Relative time shift written on the y-axis label. 

No time shift
![record_section_notshift](https://user-images.githubusercontent.com/23055374/222540568-3c43b0c5-bb73-49a0-880b-9b7c186f28e2.png)

Theoretical P-arrival alignment
![record_section_shift_on_p](https://user-images.githubusercontent.com/23055374/222540586-d7b1990a-e09a-4b37-820e-cd135015cfb1.png)

**Comment by aakash10gupta** [comment]
The new options for `time_shift_s` for aligning seismograms are very helpful. This issue can be closed now.
